libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -32,9 +44,229 @@ impl<'a> LiftAstToDml<'a> {
             }
         }
 
+        self.lift_relations(&mut schema, &mut field_ids_for_sorting);
+
+        for model in schema.models_mut() {
+            let model_name = model.name.as_str();
+            model
+                .fields
+                .sort_by_key(|field| field_ids_for_sorting[&(model_name, field.name())]);
+        }
+
         schema
     }
 
+    fn lift_relations(
+        &self,
+        schema: &mut dml::Datamodel,
+        field_ids_for_sorting: &mut HashMap<(&'a str, &'a str), ast::FieldId>,
+    ) {
+        let active_connector = self.db.active_connector();
+        let common_dml_fields = |field: &mut dml::RelationField,
```

Comment:
Also, using a closure here is kind of confusing, especially with so many parameters taken in...